File: Modulo 2/set2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdbool.h> // Adicionado para suportar o tipo bool
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");

    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    float L = (float)letters / words * 100;
    float S = (float)sentences / words * 100;

    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", grade);
    }
}

int count_letters(string text)
{
    int count = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]))
        {
            count++;
        }
    }
    return count;
}

int count_words(string text)
{
    int count = 0;
    bool in_word = false;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isspace(text[i]))
        {
            if (in_word)
            {
                count++;
                in_word = false;
            }
        }
        else
        {
            in_word = true;
        }
    }

    if (in_word)
    {
        count++;
    }

    return count;
}

int count_sentences(string text)
{
    int count = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count++;
        }
    }
    return count;
}

File: Modulo 6/set6/redability/readability.py
from cs50 import get_string
import re

def main():
    # Solicitar texto ao usuário
    text = get_string("Text: ")

    # Contar letras, palavras e frases
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # Calcular médias por 100 palavras
    L = (letters / words) * 100
    S = (sentences / words) * 100

    # Calcular índice Coleman-Liau
    index = 0.0588 * L - 0.296 * S - 15.8
    grade = round(index)

    # Determinar o nível de legibilidade
    if grade >= 16:
        print("Grade 16+")
    elif grade < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {grade}")

def count_letters(text):
    """Conta o número de letras no texto (a-z, A-Z)"""
    return len(re.findall(r'[a-zA-Z]', text))

def count_words(text):
    """Conta o número de palavras no texto (separadas por espaços)"""
    # Remove espaços extras e divide por espaços
    return len(text.strip().split())

def count_sentences(text):
    """Conta o número de frases no texto (terminadas por . ! ?)"""
    return len(re.findall(r'[.!?]', text))

if __name__ == "__main__":
    main()
